### DISCUSSION TITLE
Using general functionalities on new docs

### DISCUSSION THREAD

**Discussion by VenkataBuddha** [MaartenGr/BERTopic#973] [Q&A]
After predicting topics on new docs using transform function. How can we update or make use of the visualization methods and other topic info methods on the new docs? 

For ex:
topic_model.get_document_info(test['Body_title_abs_without_stopwords'])

WHEN I RUN THE ABOVE CODE: I get the following error: **array length 5102 does not match index length 3364**

The number 5102 is the length of my training docs and 3364 is the length of my test docs.

**Comment by MaartenGr** [comment]
The `.get_document_info` function is currently only used for documents on which the model was trained as information about unseen documents are not saved within the topic model and you would have to pass manually from the output variables.

**Reply by VenkataBuddha** [reply]
Thanks for the response!
Can you provide any sample example of what you meant by manually passing from output variables?

**Reply by MaartenGr** [reply]
In order to create similar output of `.get_document_info`, you would have to create the dataframe yourself. So let's say you did the following:

```python
predicted_topics, predicted_probs = topic_model.transform(unseen_documents)
```

then you can create a similar dataframe as follows:

```python
import pandas as pd
df = pd.DataFrame({"Document": unseen_docs, "Topic": predicted_topics, "Probability": predicted_probs})
```

If you would like to have more metadata, you can combine `df` with the output of `.get_document_info`.

**Reply by VenkataBuddha** [reply]
Is there a manual way to visualize the new documents?
like to generate a plot simialr to visualize_documents() method

**Reply by MaartenGr** [reply]
That is currently not implemented, you would have to adapt the code [here](https://github.com/MaartenGr/BERTopic/blob/master/bertopic/plotting/_documents.py) in order to do that for unseen documents.
